docs/proposals/resource-qos.md

```diff
@@ -0,0 +1,151 @@
+# Resource Quality of Service in Kubernetes
+
+**Author**: Ananya Kumar (@AnanyaKumar)
+
+**Status**: Draft proposal; prototype in progress.
+
+*This document presents the design of resource quality of service for containers in Kubernetes, and describes use cases and implementation details.*
+
+## Motivation
+
+Kubernetes allocates resources to containers in a simple way. Users can specify resource limits for containers. For example, a user can specify a 1gb memory limit for a container. The scheduler uses resource limits to schedule containers (technically, the scheduler schedules pods comprised of containers). For example, the scheduler will not place 5 containers with a 1gb memory limit onto a machine with 4gb memory. Additionally, Kubelet does not allow containers to exceed specified limits. In the current implementation, there aren’t robust mechanisms in place to ensure that containers run reliably on an overcommitted system.

```

Comment:
I would say "Currently, Kubernetes" -- otherwise it's not clear that this is about to change.